lib/streaming.js

```diff
@@ -146,29 +112,13 @@ StreamProxy.prototype.setStream = function(apiCall, argument) {
     });
   });
   if (this.type !== StreamType.CLIENT_STREAMING) {
-    stream.on('data', function() {
-      var args = Array.prototype.slice.call(arguments, 0);
-      args.unshift('data');
-      self.emit.apply(self, args);
-    });
-    // Pushing null causes an ending process of the readable stream.
-    stream.on('end', function() {
-      self.push(null);
-    });
-    // This is required in case no 'data' handler exists.
-    this.resume();
+    this.setReadable(stream);
   }
   if (this.type !== StreamType.SERVER_STREAMING) {
-    this._writeQueue.forEach(function(data) {
-      stream.write(data);
-    });
-    this._writeQueue = [];
-    if (this._isEndCalled) {
-      stream.end();
-    }
-    if (this._isCancelCalled) {
-      stream.cancel();
-    }
+    this.setWritable(stream);
+  }
+  if (this._isCancelCalled) {
```

Comment:
Does `_isCancelCalled` get set somewhere else?

Reply:
Whoops, I removed it from the constructor, but I think it should have stayed :)

Reply:
But yes, if you look at StreamProxy#cancel, we check to see if the stream has been set yet, if not we set this flag to true so we can cancel it straight away.